src/olympia/abuse/utils.py

```diff
@@ -0,0 +1,78 @@
+from olympia import amo
+from olympia.activity import log_create
+from olympia.addons.models import Addon
+from olympia.users.models import UserProfile
+
+
+class CinderAction:
+    description = 'Action has been taken'
+
+    def __init__(self, cinder_report):
+        self.cinder_report = cinder_report
+        self.abuse_report = cinder_report.abuse_report
+
+    def process(self):
+        raise NotImplementedError
+
+    def notify_targets(self, targets):
+        # TODO: notify target
+        pass
+
+    def notify_reporter(self):
+        if self.abuse_report.reporter or self.abuse_report.reporter_email:
+            # TODO: notify reporter
+            pass
+
+
+class CinderActionBanUser(CinderAction):
+    description = 'Account has been banned'
+
+    def process(self):
+        if user := self.abuse_report.user:
+            log_create(amo.LOG.ADMIN_USER_BANNED, user)
+            UserProfile.ban_and_disable_related_content_bulk([user], move_files=True)
+            self.notify_reporter()
+            self.notify_targets([user])
+
+
+class CinderActionDisableAddon(CinderAction):
+    description = 'Add-on has been disabled'
+
+    def process(self):
+        addon = Addon.unfiltered.filter(guid=self.abuse_report.guid).first()
+        if addon and addon.status != amo.STATUS_DISABLED:
+            addon.force_disable()
+            self.notify_reporter()
+            self.notify_targets(addon.authors.all())
+
+
+class CinderActionEscalateAddon(CinderAction):
+    def process(self):
+        from olympia.reviewers.models import NeedsHumanReview
+
+        addon = Addon.unfiltered.filter(guid=self.abuse_report.guid).first()
+        if addon:
+            addon.set_needs_human_review_on_latest_versions(
```

Comment:
Could this be the wrong version(s) ? Do we care ?

Reply:
I guess if we have the version (which we won't necessarily have) we could specifically flag that version...